mlrun/datastore/store_resources.py

```diff
@@ -165,3 +165,27 @@ def get_store_resource(uri, db=None, secrets=None, project=None):
     else:
         stores = mlrun.store_manager.set(secrets, db=db)
         return stores.object(url=uri)
+
+
+def allow_empty_targets():
+    """
+    Modifier function to allow resource (feature vector/set) uris without material targets
+
+    Usage::
+
+        train = train_op(...)
+        train.apply(allow_empty_targets())
+        train.run(inputs={"data": vector.uri})
+    """
+
+    def _allow_empty_targets(task):
+        from os import environ
+
+        from kubernetes import client as k8s_client
+
+        environ["ALLOW_EMPTY_RESOURCE_TARGET"] = "1"  # for local runs
```

Comment:
This will cause unexpected behavior, since for local runs it will set the env variable in the client's env, and it will hold for any run that is done locally, not just for runs of the runtime for which the `apply()` is called. This is not what the user intended.

Reply:
i believe its better to have false positive here than false negative, if we set its just suppressing the exception , if its not turned on when u need it you wont figure out why the apply() didnt work